tools/google-cloud-support-slackbot/google_cloud_support_slackbot.py

```diff
@@ -0,0 +1,811 @@
+    user_id = data.get('user_id')
+    user_name = data.get('user_name')
+    user_inputs = data.get('text').split(' ', 1)
+    command = user_inputs[0]
+
+    if command == 'track-case':
+        try:
+            case = user_inputs[1]
+            track_case(channel_id, channel_name, case, user_id)
+        except Exception as e:
+            logging.error(e, ' : {}'.format(datetime.now()))
+            client.chat_postEphemeral(channel=channel_id, user=user_id, text="The track-case command expects argument [case_number]. The case number provided did not match with any cases in your org")
+
+    elif command == 'add-comment':
+        try:
+            parameters = user_inputs[1].split(' ', 1)
+            case = parameters[0]
+            comment = parameters[1]
+            p = Process(target=add_comment, args=(channel_id, case, comment, user_id, user_name,))
+            p.start()
+        except Exception as e:
+            logging.error(e, ' : {}'.format(datetime.now()))
+            client.chat_postEphemeral(channel=channel_id, user=user_id, text="The add-comment command expects arguments [case_number] [comment]. The comment does not need to be encapsulated in quotes. Your case number did not match with any cases in your org.")
+
+    elif command == 'change-priority':
+        try:
+            parameters = user_inputs[1].split(' ', 1)
+            case = parameters[0]
+            priority = parameters[1]
+            p = Process(target=change_priority, args=(channel_id, case, priority, user_id,))
+            p.start()
+        except Exception as e:
+            logging.error(e, ' : {}'.format(datetime.now()))
+            client.chat_postEphemeral(channel=channel_id, user=user_id, text="The change-priority command expects arguments [case_number] [priority, must be either P1|P2|P3|P4]. Your case number did not match with any cases in your org, or the priority did not match the expected values.")
+
+    #elif command == 'escalate':
+        #try:
+            #parameters = user_inputs[1].split(' ', 2)
+            #case = parameters[0]
+            #reason = parameters[1]
+            #justification = parameters[2]
+            #p = Process(target=escalate, args=(channel_id, case, user_id, reason, justification, user_name))
+            #p.start()
+        #except Exception as e:
+            #client.chat_postEphemeral(channel=channel_id, user=user_id, text="The escalate command expects arguments [reason, must be either REASON_UNSPECIFIED|RESOLUTION_TIME|TECHNICAL_EXPERTISE|BUSINESS_IMPACT] [justification]. The justification does not need to be encapsulated in quotes. Either your case number did not match with any cases in your org, the reason did not match one of the expected values, or the justification was missing")
+
+    elif command == 'stop-tracking':
+        try:
+            case = user_inputs[1]
+            stop_tracking(channel_id, channel_name, case, user_id)
+        except Exception as e:
```

Comment:
What specific exception are you capturing here? Please narrow it down to the specific situations that you are trying to handle, this is too broad.